File: Data Structures & Algorithms/duplicate-integer/submission-0.py
class Solution:
    def hasDuplicate(self, nums: List[int]) -> bool:
        dct = {}
        for i in nums:
            if i in dct:
                dct[i] += 1
            else:
                dct[i] = 1
        for i in dct.values():
            # print(i)
            if i >= 2:
                return True
        return False

File: Data Structures & Algorithms/binary-search/submission-0.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} target
     * @return {number}
     */
    binary_search(nums, target, l, r) {
        if ( l > r ) return -1;
        let  m = l + Math.floor((r-l)/2)
        if (nums[m] === target) return m;
        return (nums[m] < target) ?
            this.binary_search(nums, target, m+1, r) :
            this.binary_search(nums, target, l, m-1);
    }

    search(nums, target) {
        return this.binary_search(nums, target, 0, nums.length - 1);
    }
}